core/field_variable.js

```diff
@@ -143,11 +171,16 @@ Blockly.FieldVariable.dropdownCreate = function() {
   if (this.sourceBlock_) {
     workspace = this.sourceBlock_.workspace;
   }
-
   if (workspace) {
+    var variableTypes = this.getVariableTypes_();
+    var variableModelList = [];
     // Get a copy of the list, so that adding rename and new variable options
     // doesn't modify the workspace's list.
-    variableModelList = workspace.getVariablesOfType('');
+    for (var i = 0, variableType; i < variableTypes.length; i++) {
```

Comment:
Yes please, it should be initialized inside the loop if you're not using it in the for condition.